ArduCopter/AP_RCSwitch.cpp

```diff
@@ -0,0 +1,592 @@
+                    // high switch maintains current position
+                    copter.g2.winch.release_length(0.0f);
+                    copter.Log_Write_Event(DATA_WINCH_LENGTH_CONTROL);
+                    break;
+                default:
+                    // all other position relax winch
+                    copter.g2.winch.relax();
+                    copter.Log_Write_Event(DATA_WINCH_RELAXED);
+                    break;
+                }
+#endif
+            break;
+
+        case WINCH_CONTROL:
+#if WINCH_ENABLED == ENABLED
+            switch (ch_flag) {
+                case LOW:
+                    // raise winch at maximum speed
+                    copter.g2.winch.set_desired_rate(-copter.g2.winch.get_rate_max());
+                    break;
+                case HIGH:
+                    // lower winch at maximum speed
+                    copter.g2.winch.set_desired_rate(copter.g2.winch.get_rate_max());
+                    break;
+                case MIDDLE:
+                    copter.g2.winch.set_desired_rate(0.0f);
+                    break;
+                }
+#endif
+            break;
+
+        case RC_OVERRIDE_ENABLE:
+            // Allow or disallow RC_Override
+            switch (ch_flag) {
+                case HIGH: {
+                    copter.ap.rc_override_enable = true;
+                    break;
+                }
+                case MIDDLE:
+                    // nothing
+                    break;
+                case LOW: {
+                    copter.ap.rc_override_enable = false;
+                    break;
+                }
+            }
+            break;
+    default:
+        AP_RCSwitch::do_aux_function(ch_option, ch_flag);
+        break;
+    }
+}
+
+// save_trim - adds roll and pitch trims from the radio to ahrs
+void Copter::save_trim()
+{
+    // save roll and pitch trim
+    float roll_trim = ToRad((float)channel_roll->get_control_in()/100.0f);
+    float pitch_trim = ToRad((float)channel_pitch->get_control_in()/100.0f);
+    ahrs.add_trim(roll_trim, pitch_trim);
+    Log_Write_Event(DATA_SAVE_TRIM);
+    gcs().send_text(MAV_SEVERITY_INFO, "Trim saved");
+}
+
+// auto_trim - slightly adjusts the ahrs.roll_trim and ahrs.pitch_trim towards the current stick positions
+// meant to be called continuously while the pilot attempts to keep the copter level
+void Copter::auto_trim()
+{
+    if (auto_trim_counter > 0) {
+        auto_trim_counter--;
+
+        // flash the leds
+        AP_Notify::flags.save_trim = true;
+
+        // calculate roll trim adjustment
+        float roll_trim_adjustment = ToRad((float)channel_roll->get_control_in() / 4000.0f);
+
+        // calculate pitch trim adjustment
+        float pitch_trim_adjustment = ToRad((float)channel_pitch->get_control_in() / 4000.0f);
+
+        // add trim to ahrs object
+        // save to eeprom on last iteration
+        ahrs.add_trim(roll_trim_adjustment, pitch_trim_adjustment, (auto_trim_counter == 0));
```

Comment:
something looks wrong , in save_trim() we get_control_in and divide by 100, here we get control_in and divide by 4000